File: apps/mobile/lib/features/profile/data/dtos/update_user_preferences_request_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'update_user_preferences_request_dto.g.dart';

@JsonSerializable(createFactory: false, includeIfNull: false)
class UpdateUserPreferencesRequestDto {
  const UpdateUserPreferencesRequestDto({
    this.currencyCode,
    this.locale,
    this.country,
  });

  final String? currencyCode;
  final String? locale;
  final String? country;

  Map<String, dynamic> toJson() =>
      _$UpdateUserPreferencesRequestDtoToJson(this);
}

File: apps/api/FinanceControl.Shared/Dtos/Request/UpdateUserPreferencesRequestDto.cs
namespace FinanceControl.Shared.Dtos.Request
{
    public class UpdateUserPreferencesRequestDto
    {
        public string? CurrencyCode { get; set; }
        public string? Locale { get; set; }
        public string? Country { get; set; }
        public string? AnalyticsConfig { get; set; }
    }
}
